azure/converters/image.go

```diff
@@ -68,6 +71,29 @@ func specificImageToSDK(image *infrav1.Image) (*compute.ImageReference, error) {
 	}, nil
 }
 
+func computeImageToSDK(image *infrav1.Image) (*compute.ImageReference, error) {
+	if image.ComputeGallery.ResourceGroup == nil {
+		return &compute.ImageReference{
+			CommunityGalleryImageID: to.StringPtr(fmt.Sprintf("/CommunityGalleries/%s/Images/%s/Versions/%s",
+				image.ComputeGallery.Gallery,
+				image.ComputeGallery.Name,
+				image.ComputeGallery.Version)),
+		}, nil
+	}
+
+	idTemplate := "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/galleries/%s/images/%s/versions/%s"
```

Comment:
can we reuse `sigImageToSDK` for this?

Reply:
We would have to map the parameters from `image.ComputeGallery` to `image.SharedGallery`, which I'd rather not do, I'd convert other way around. So I would either do:
```go
       if image.SharedGallery != nil {
               return computeImageToSDK(sigToCompute(image))
       if image.ComputeGallery != nil {
               return computeImageToSDK(image)
       }
```

or handle the conversion in `computeImageToSDK` and do:
```go
       if image.ComputeGallery != nil || image.SharedGallery != nil {
               return computeImageToSDK(image)
       }
...

func computeImageToSDK(image *infrav1.Image) (*compute.ImageReference, error) {
  idTemplate := "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/galleries/%s/images/%s/versions/%s"

  if image.SharedGallery != nil {
    return &compute.ImageReference{
      ID: to.StringPtr(fmt.Sprintf(idTemplate,
        image.SharedGallery.SubscriptionID,
        image.SharedGallery.ResourceGroup,
        image.SharedGallery.Gallery,
        image.SharedGallery.Name,
        image.SharedGallery.Version,
      )),
    }, nil
  }

  if image.ComputeGallery.ResourceGroup != nil {
    return &compute.ImageReference{
      ID: to.StringPtr(fmt.Sprintf(idTemplate,
        image.ComputeGallery.SubscriptionID,
        image.ComputeGallery.ResourceGroup,
        image.ComputeGallery.Gallery,
        image.ComputeGallery.Name,
        image.ComputeGallery.Version,
      )),
    }, nil
  }

  return &compute.ImageReference{
    CommunityGalleryImageID: to.StringPtr(fmt.Sprintf("/CommunityGalleries/%s/Images/%s/Versions/%s",
      image.ComputeGallery.Gallery,
      image.ComputeGallery.Name,
      image.ComputeGallery.Version)),
  }, nil
}
```

I'll push the 2nd approach to continue the discussion.